src/services/breakpoints.ts

```diff
@@ -31,7 +31,8 @@ namespace ts.BreakpointResolver {
         }
 
         // Cannot set breakpoint in ambient declarations
-        if (isInAmbientContext(tokenAtLocation)) {
+        // (Tokens do not have flags set, but their parents do.)

```

Comment:
This is surprising, wouldn't this result in breaks in the incremental parser?

Reply:
We call `parseTokenNode` for tokens. `parseTokenNode` in turn calls `finishNode`, which should set these flags. If a token doesn't have the correct context then we should investigate why.